cockatrice/src/dlg_settings.cpp

```diff
@@ -464,25 +464,154 @@ void UserInterfaceSettingsPage::retranslateUi()
+    lpSpoilerGrid->addWidget(mpSpoilerPathButton, 1, 2);
+    lpSpoilerGrid->addWidget(&mcNextUpdateTimeToolTipLabel, 2, 0);
+    lpSpoilerGrid->addWidget(&mcNextUpdateTimeLabel, 2, 1);
+
+    lpSpoilerGrid->setColumnStretch(0, 2);
+    lpSpoilerGrid->setColumnStretch(1, 4);
+    lpSpoilerGrid->setColumnStretch(2, 1);
+
+    lpGeneralGrid->addWidget(&mcGeneralMessageLabel, 0, 0);
+
+    // On a change to the check box, hide/unhide the other fields
+    // On a change to the combo box, send a signal to the timer threads
+    connect(&mcDownloadSpoilersCheckBox, SIGNAL(stateChanged(int)), settingsCache, SLOT(setDownloadSpoilerStatus(int)));
+    connect(&mcDownloadSpoilersCheckBox, SIGNAL(stateChanged(int)), this, SLOT(setSpoilersEnabled(int)));
+    connect(&msDownloadSpoilersTimeIntervalComboBox, SIGNAL(currentTextChanged(QString)), this, SLOT(setDownloadSpoilerTime(QString)));
+
+    mpGeneralGroupBox = new QGroupBox;
+    mpGeneralGroupBox->setLayout(lpGeneralGrid);
+
+    mpSpoilerGroupBox = new QGroupBox;
+    mpSpoilerGroupBox->setLayout(lpSpoilerGrid);
+
+    QVBoxLayout *lpMainLayout = new QVBoxLayout;
+    lpMainLayout->addWidget(mpGeneralGroupBox);
+    lpMainLayout->addWidget(mpSpoilerGroupBox);
+
+    setLayout(lpMainLayout);
+}
+
+void DeckEditorSettingsPage::spoilerPathButtonClicked()
+{
+    QString lsPath = QFileDialog::getExistingDirectory(this, tr("Choose path"));
+    if (! lsPath.isEmpty())
+    {
+        mpSpoilerSavePathLineEdit->setText(lsPath + "/spoiler.xml");
+        settingsCache->setSpoilerDatabasePath(lsPath + "/spoiler.xml");
+    }
+}
+
+void DeckEditorSettingsPage::setDownloadSpoilerTime(QString asValue)
+{
+    // Sets the downloadspoilerstimeMinutes field (stored in MINUTES (int))
+    QMap<int, QString> lacSettingTimes = settingsCache->getDownloadSpoilerTimeIntervals();
+    int lnMinutesToReload = lacSettingTimes.key(asValue);
+
+    // Update settings cache downloadspoilerstimeMinutes
+    settingsCache->setDownloadSpoilerTimeMinutes(lnMinutesToReload);
+
+    // Update the GUI display
+    updateDownloadTimer();
+}
+
+void DeckEditorSettingsPage::setSpoilersEnabled(int anInput)
+{
+    bool lbVisibility = (anInput != 0);
+
+    msDownloadSpoilersLabel.setEnabled(lbVisibility);
+    msDownloadSpoilersTimeIntervalComboBox.setEnabled(lbVisibility);
+    mcSpoilerSaveLabel.setEnabled(lbVisibility);
+    mpSpoilerSavePathLineEdit->setEnabled(lbVisibility);
+    mpSpoilerPathButton->setEnabled(lbVisibility);
+    updateDownloadTimer(lbVisibility);
+}
+
+void DeckEditorSettingsPage::updateDownloadTimer(bool abIsEnabled)
+{
+    // Update the display so the user sees when the next download time is or nothing if disabled
+    if (abIsEnabled)
+    {
+        long lnTimeUntilNextUpdate = QDateTime::currentMSecsSinceEpoch() - settingsCache->getDownloadSpoilerLastUpdateTime();
+        float lnTimeToWaitBetween = settingsCache->getDownloadSpoilerTimeMinutes() * 60000;
```

Comment:
Why float? Use long, long long (or, better, qint64) and avoid the static_cast<long> in the following lines